package.json

```diff
@@ -77,15 +77,16 @@
     "test": "npm run typetest && nyc --reporter=none ava --timeout=2m test && nyc report --reporter=html --reporter=text",
     "report-coverage": "nyc report --reporter=text-lcov > coverage.lcov && codecov",
     "static": "eslint index.js lib src test",
-    "build": "npm run static && npm run docs:build && gulp",
+    "build": "npm run static && gulp",
     "instrument": "nyc instrument ./src ./instrumented",
     "postinstrument": "node test/helpers/retarget-browser-coverage.js",
-    "docs": "npm run docs:build && npm run docs:publish",
-    "docs:build": "rimraf docs && jsdoc -c .jsdoc.json && node scripts/readme-toc.js",
-    "docs:publish": "npm run docs:build && echo nodecg.com >> docs/CNAME && gh-pages -d docs -s \"**\"",
+    "docs:buildonly": "node scripts/docs-build-one.js",
+    "docs:latest": "node scripts/docs-build-one.js latest",
+    "docs:tag": "node scripts/docs-build-one.js tag",
+    "docs:all": "node scripts/docs-build-all.js",
```

Comment:
Can you walk me through what each of these scripts does? And are they all safe to run locally, or will they commit/tag things on git?

Reply:
* `docs:all`/`docs-build-all.js`: build every version of the docs we need, as well as for latest tip-of-tree, setup CNAME and (pending) index file, and push to origin
* `docs:buildonly`/`docs-build-one.js`: build docs for latest tip-of-tree
* `docs:latest`/`docs-build-one.js latest`: build docs for latest tip-of-tree, and publish to the `latest` section of the docs
* `docs:tag`/`docs-build-one.js tag`: build docs for latest tip-of-tree, and publish to corresponding section of docs if environment variable `TRAVIS_TAG` is set and valid (otherwise will just build)

Scripts that use `docs-build-one.js` are designed to be run in CI (using `GITHUB_TOKEN`), but can be used locally if you have push access to origin (otherwise will fail to publish, but should still build).

`docs-build-all.js` must be run locally with push access to origin, as it theoretically only needs to run once.